libs/providers/flagd/src/lib/service/grpc/service.ts

```diff
@@ -219,9 +219,9 @@ export class GRPCService implements Service {
     parser?: (struct: Struct) => PbJsonValue
   ): Promise<ResolutionDetails<T>> {
     if (this._cacheActive) {
-      const cached: ResolutionDetails<T> | undefined = this._cache?.get(flagKey);
+      const cached = this._cache?.get(flagKey);
       if (cached) {
-        return { ...cached, reason: StandardResolutionReasons.CACHED };
+        return { ...cached, reason: StandardResolutionReasons.CACHED } as ResolutionDetails<T>;
```

Comment:
I had to explicitly cast to `ResolutionDetails<T>` because the cache only knows that it's `ResolutationDetails<FlagValue>`.